pkg/controller/cincinnati/cincinnati_controller_test.go

```diff
@@ -100,8 +148,215 @@ func TestEnsureEnvConfig(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	assert.Equal(t, found.ObjectMeta.OwnerReferences[0].Name, cincinnati.Name)
-	assert.NotNil(t, found.Data["pe.upstream"])
+	verifyOwnerReference(t, found.ObjectMeta.OwnerReferences[0], cincinnati)
+	assert.NotEmpty(t, found.Data["pe.upstream"])
+}
+
+func TestEnsureAdditionalTrustedCA(t *testing.T) {
+	tests := []struct {
+		name          string
+		existingObjs  []runtime.Object
+		expectedError error
+	}{
+		{
+			name: "NoImage",
+			existingObjs: []runtime.Object{
+				newDefaultCincinnati(),
+			},
+			expectedError: nil,
+		},
+		{
+			name: "NoAdditionalTrustedCAName",
+			existingObjs: []runtime.Object{
+				newDefaultCincinnati(),
+				func() *configv1.Image {
+					image := newImage()
+					image.Spec.AdditionalTrustedCA.Name = ""
+					return image
+				}(),
+			},
+			expectedError: nil,
+		},
+		{
+			name: "NoConfigMap",
+			existingObjs: []runtime.Object{
+				newDefaultCincinnati(),
+				newImage(),
+			},
+			expectedError: fmt.Errorf("configmaps \"%v\" not found", newImage().Spec.AdditionalTrustedCA.Name),
+		},
+		{
+			name: "ConfigMapCreate",
+			existingObjs: []runtime.Object{
+				newDefaultCincinnati(),
+				newImage(),
+				newConfigMap(),
+			},
+			expectedError: nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			cincinnati := newDefaultCincinnati()
+			r := newTestReconciler(test.existingObjs...)
+
+			resources, err := newKubeResources(cincinnati, testOperandImage)
+			err = r.ensureAdditionalTrustedCA(context.TODO(), log, cincinnati, resources)
+
+			verifyError(t, err, test.expectedError)
+
+			if test.name == "ConfigMapCreate" {
+				found := &corev1.ConfigMap{}
```

Comment:
> I don't see a specific reason why we need 3 versions of TestEnsureAdditionalTrustedCA

In a success scenario, you may want to verify more things like resources are created properly, that you typically may not care about in a failure scenario. Like In this particular case, I ended up with a  if-condition to handle that case. Getting rid of that if-condition would need expectedConfigMap object to be passed to the struct, which means I'd have to construct that object. Just trying to find the right balance here.